src/datachain/lib/signal_schema.py

```diff
@@ -69,6 +75,29 @@ def __init__(self, method: str, field):
         )
 
 
+dynamic_feature_models: dict[str, type[BaseModel]] = {}
```

Comment:
should it be `_dynamic_feature_models` (protected) or even `__dynamic_feature_models`?

do we use it as a cache?

Reply:
How do this and `custom_types` differ from the `ModelStore`? Is it possible to reuse `ModelStore` for either of these?

Reply:
Sure, changed it to `_dynamic_feature_models` (protected) and yes, this is used as a cache.

This is intended more as a cache of the defined dynamic models, instead of the registered list of `ModelStore` - as this only includes the created dynamic models, not all models. I'm not 100% clear on how `ModelStore` is used and what caveats there would be with using it for this cache, especially since it does not have any mechanism for creating new models, only registering existing ones. I could look into integrating these two features together if you think it would be useful, but I'm not sure how complex that would be (and what side-effects there might be) and so it should probably be done in a followup.

Reply:
Have you tried just dropping `_dynamic_feature_models` and seeing if it makes any difference? AFAIU because you use `DataModel` as the base when creating the dynamic models, they should already be registered to the `ModelStore` and should already be cached and should be caught [here](https://github.com/iterative/datachain/blob/2a86ad17b910d39e197ea24e3fd4a507b62249d0/src/datachain/lib/signal_schema.py#L227-L229) before getting duplicated.